csrc/tensor_view.cpp

```diff
@@ -49,6 +49,21 @@ TensorView::TensorView(
 
 NVFUSER_DEFINE_CLONE(TensorView)
 
+bool TensorView::sameDefinition(const Val* other) const {
+  // Val::sameDefinition checks nullptr, dtype, vtype, and definition.
+  if (!Val::sameDefinition(other)) {
+    return false;
+  }
+  if (!other->isA<TensorView>()) {
```

Comment:
do we need to check this, since other has already passed Val::sameDefinition, which checks typeid already.